Source/WebKit/Shared/skia/CoreIPCSkString.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if USE(SKIA)
+
+#include <skia/core/SkString.h>
+
+namespace WebKit {
+
+class CoreIPCSkString {
+public:
+    CoreIPCSkString(const SkString& string)
+        : m_string(string)
+    {
+    }
+
+    static SkString create(std::span<const char> span)
+    {
+        return { span.data(), span.size() };
+    }
+
+    std::span<const char> data() const
+    {
+        return { m_string.data(), m_string.size() };
+    }
+
+private:
+    const SkString& m_string;
```

Comment:
Storing an unsafe reference isn't ideal.  Are the Skia maintainers amenable to adding a span accessor and constructor to SkString?

Reply:
I don't think Skia adopts C++20 `std::span` in their public API.
They have `SkSpan` instead.
https://github.com/WebKit/WebKit/blob/main/Source/ThirdParty/skia/include/private/base/SkSpan_impl.h
Should I ask the upstream Skia to add a `SkString` constructor which takes `SkSpan<const char>`?

Reply:
I think raw reference is inevitable for lvalue-reference-taking encoders of external types for performance.